src/link-utils.js

```diff
@@ -0,0 +1,35 @@
+const parseRawLinks = (rawLinks) => {
+  if (typeof rawLinks !== "string") {
+    return [];
+  }
+
+  let links = [];
+  const splitLinks = rawLinks.split(",");
+
+  for (var i = 0; i < splitLinks.length; i++) {
+    let link = {};
+    const splitParts = splitLinks[i].split(";");
+
+    link.href = splitParts[0].replace(/<(.*)>/, "$1").trim();
+    link.rel = splitParts[1].replace(/rel="(.*)"/, "$1").trim();
+    link.title = splitParts[2].replace(/title="(.*)"/, "$1").trim();
+
+    links.push(link);
+  }
+
+  return links;
+};
+
+const resolveUrls = (links, baseUrl) => {
+  const linksCopy = Object.assign([], links);
+
+  for (var i = 0; i < linksCopy.length; i++) {
```

Comment:
consider trying to use for...in loops. I think it's cleaner than the traditional for loops minus a few cases where the arrays aren't so straight forward

Reply:
So I tried using the for-in loop, but it doesn't seem to be getting the objects in my `linksCopy` array correctly. Here is the change I made to the code:

![screen shot 2018-04-12 at 9 39 13 am](https://user-images.githubusercontent.com/13635808/38691385-93a8c4ec-3e35-11e8-9b20-965fdc57a757.png)

And here is one of the failing tests:

![screen shot 2018-04-12 at 9 39 24 am](https://user-images.githubusercontent.com/13635808/38691398-9cce3656-3e35-11e8-8571-27d18efe6d1d.png)

Reply:
Ok, I used the `for...of` loop